### DISCUSSION TITLE
Front end  and Back end languages and framework recommendations

### DISCUSSION THREAD

**Discussion by VSRohod** [community/community#65704] [Other Feature Feedback, Questions, & Ideas] [answered]
I am a beginner programmer and wanted to know what is being most used in the market, what languages and frameworks both for Front End and for Back End

**Comment by arafajs** [comment, marked as answer]
@VSRohod 
**Front-end**

**1. JavaScript:** JavaScript is the most popular programming language for the web, and it is used to create interactive and dynamic web pages. Some popular JavaScript frameworks include React, Angular, and Vue.js.
**2. HTML:** HTML is the markup language used to define the structure of a web page. It is used to create the basic elements of a web page, such as headings, paragraphs, and images.
**3. CSS:** CSS is the style sheet language used to control the appearance of a web page. It is used to define the colors, fonts, and layout of a web page.

**Back-end**

**1. Python:** Python is a general-purpose programming language that is popular for back-end development. It is known for its simplicity and readability, and it has a large and active community of developers. Some popular Python frameworks include Django, Flask, and Pyramid.
**2. Java:** Java is another popular general-purpose programming language that is used for back-end development. It is known for its portability and scalability, and it is used by many large companies. Some popular Java frameworks include Spring Boot, Micronaut, and Quarkus.
**3. Node.js:** Node.js is a JavaScript runtime environment that can be used for both front-end and back-end development. It is known for its asynchronous programming model, which makes it ideal for building scalable and performant applications. Some popular Node.js frameworks include Express, NestJS, and Sails.js.

**Reply by VSRohod** [reply]
Oh thank you so much ✨

**Comment by idarbandi** [comment]
take a look at this 
[maktab](https://github.com/idarbandi/maktab)
Might be a good help for you
 maybe stars
